sklego/meta/_grouped_utils.py

```diff
@@ -45,13 +46,15 @@ def _split_groups_and_values(
     _shape_check(X, min_value_cols)
 
     try:
+        lgroups = as_list(groups)
```

Comment:
I guess we should set `group_sizes` to also be a non-list type? Bit of a nit this one. 

Also: maybe `groups_list` instead of `lgroups`.